hack/dind

```diff
@@ -65,7 +65,7 @@ if [ -f /sys/fs/cgroup/cgroup.controllers ]; then
 	xargs -rn1 < /sys/fs/cgroup/cgroup.procs > /sys/fs/cgroup/init/cgroup.procs || :
 	# enable controllers
 	sed -e 's/ / +/g' -e 's/^/+/' < /sys/fs/cgroup/cgroup.controllers \
-		> /sys/fs/cgroup/cgroup.subtree_control
+		> /sys/fs/cgroup/cgroup.subtree_control || :
```

Comment:
https://github.com/moby/moby/pull/48850#issuecomment-2480194981

```suggestion
	# this happens in a loop because things like "docker exec" on our dind
	# container will create new processes, which creates a race between our
	# moving everything to "init" and enabling subtree_control
	while ! {
		# move the processes from the root group to the /init group,
		# otherwise writing subtree_control fails with EBUSY.
		# An error during moving non-existent process (i.e., "cat") is ignored.
		xargs -rn1 < /sys/fs/cgroup/cgroup.procs > /sys/fs/cgroup/init/cgroup.procs || :
		# enable controllers
		sed -e 's/ / +/g' -e 's/^/+/' < /sys/fs/cgroup/cgroup.controllers \
			> /sys/fs/cgroup/cgroup.subtree_control
	}; do true; done
```

(I'm happy to update this PR on your behalf if you'd prefer! :heart:)

Reply:
Awesome. Thank you! @JSchltggr can you add this change? Otherwise, I can also open a new merge request.

Reply:
Thanks a lot tianon!
Unfortunately it was not possible to commit the suggestion directly. I've added the changes to another commit.

Reply:
It's probably ok to squash the 2 commits if possible to keep a cleaner git history

Reply:
@thaJeztah can you not just squash merge?

Reply:
We disabled  squash merge on this repository as it discards GPG signing of  the  PR author, and  generally prefer the contributor to squash commits  (or organise PRs into logical commits, which is part of the review process), so if @JSchltggr is able to do a squash, that'd be great, otherwise I can manually  squash and push to the branch through my maintainer permssions